labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise01.scala

```diff
@@ -99,7 +89,10 @@ object ListManipulationExercise01 {
    * Implement it whatever way suites you best. Hint: it can be done in a neat way using recursion. 
    */
   def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+    if (l == List[T]())
+      return List(l)
+    else
+      return List(l) ::: tails(l.tail)
```

Comment:
I believe that pattern-matching is considered the more idiomatic way to write this kind of thing. Of course, we didn't know about pattern matching while writing this so that's just retrospective.